src/Mobs/Monster.cpp

```diff
@@ -1176,33 +1179,44 @@ std::unique_ptr<cMonster> cMonster::NewMonsterFromType(eMonsterType a_MobType)
 
 			return cpp14::make_unique<cHorse>(HorseType, HorseColor, HorseStyle, HorseTameTimes);
 		}
+		case mtZombieVillager:
+		{
+			int Profession = Random.RandInt(6);
+			if (Profession == 6)
+			{
+				// Give farmers a better chance of spawning
+				Profession = 0;
+			}
+
+			return cpp14::make_unique<cZombieVillager>(static_cast<cVillager::eVillagerType>(Profession));
```

Comment:
I would prefer a seperate function, which handles the conversation from integer to enum. This would allow adding asserts to check, if the integer is out of range (which would be UB with a `static_cast`).

Reply:
Or better yet, a function that just returns a random profession to factor out from the `mtVillager` case above.